### DISCUSSION TITLE
Select Range in custom command.

### DISCUSSION THREAD

**Discussion by CabalCrow** [jesseduffield/lazygit#3673] [Q&A]
Can the selected files via select-range be used in a custom command?

**Comment by stefanhaller** [comment]
No, we don't expose the selection range to the template machinery. It's also a little unclear to me how that would work; you could imagine something like `{{.SelectedFiles}}`, but would that result in a string with all the files concatenated, with spaces in between? What about quoting of spaces or other special characters in filenames?

Or would it return an array, so users would have to loop over it somehow using `{{range .SelectedFiles}}`?

What would you intend to use it for, and how would you imagine it to work? That could inform how it needs to be designed.

**Reply by CabalCrow** [reply]
I run some reset & edit commands for multiple files. One way to could work is to call the external command for every number of the selected arguments, and for each call of the external command it uses the same syntax as if you were selecting only an individual file. This way custom commands can be easily converted to accept selections via an option, without editing the external command.

The other solution would be going through an array, which can be useful, but I don't have an immidiate usecase for that (albeit it can also deal with my current usecase).
